docs/quickstart.md

```diff
@@ -0,0 +1,760 @@
+------------------------------------------------------------------------------------------------------------------------------------------
+Successfully created/updated stack - sam-app in eu-central-1
+```
+This command will build package and deploy it to your AWS Account. Make sure you configure AWS credentials beforehand. In the outputs section you find API Url you can issue request against.
+!!! Info
+    To learn more about SAM deployment mechanism follow this [link](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-cli-command-reference-sam-deploy.html).
+
+## API Gateway router
+Let's say you have been asked to create another method that respond with caller name under `/hello/<name>` URL path.
+One approach would be to create another lambda with required method and configure API gateway to call it.
+=== "app_name.py"
+
+    ```python
+    def hello_name(name):
+        return {"statusCode": 200, "body": {"message": f"hello {name}!"}}
+
+
+    def lambda_handler(event, context):
+        name = event["pathParameters"]["name"]
+        return hello_name(name)
+    ```
+
+=== "template.yaml"
+
+    ```yaml hl_lines="22-33"
+    AWSTemplateFormatVersion: "2010-09-09"
+    Transform: AWS::Serverless-2016-10-31
+    Description: > 
+        Sample SAM Template for powertools-quickstart
+    Globals:
+        Function:
+            Timeout: 3
+    Resources:
+        HelloWorldFunction:
+            Type: AWS::Serverless::Function
+            Properties:
+            CodeUri: hello_world/
+            Handler: app.lambda_handler
+            Runtime: python3.9
+            Events:
+                HelloWorld:
+                Type: Api 
+                Properties:
+                    Path: /hello
+                    Method: get
+        HelloWorldFunctionName:
+            Type: AWS::Serverless::Function
+            Properties:
+            CodeUri: hello_world/
+            Handler: app_name.lambda_handler
+            Runtime: python3.9
+            Events:
+                HelloWorldName:
+                Type: Api
+                Properties:
+                    Path: /hello/{name}
+                    Method: get
+    Outputs:
+        HelloWorldApi:
+            Description: "API Gateway endpoint URL for Prod stage for Hello World function"
+            Value: !Sub "https://${ServerlessRestApi}.execute-api.${AWS::Region}.amazonaws.com/Prod/hello/"
+    ```
+
+This way definitively works. The downside of this approach is that you need to create new lambda file for every method or path and also configure SAM template to add new lambda configuration and configure API Gateway to point into them. For small additional methods it might be too much boilerplate code.  
+Another approach though that might be more suitable in this case is to have all of our methods under control of one lambda and "route" into specific method based on a request. 
```

Comment:
This can be much shorter and straight to the point: If we have many methods we should use routing.

Reply:
added based on suggestion